actionmailer/lib/action_mailer/mail_delivery_job.rb

```diff
@@ -9,7 +9,10 @@ module ActionMailer
   #
   # Exceptions are rescued and handled by the mailer class.
   class MailDeliveryJob < ActiveJob::Base # :nodoc:
-    queue_as { ActionMailer::Base.deliver_later_queue_name }
+    queue_as do
+      mailer_class = arguments.first.constantize
```

Comment:
Should not we be calling the `mailer_class` method here?

Reply:
Yes, you're right. The [comment](https://github.com/rails/rails/blob/fa17c9a/actionmailer/lib/action_mailer/mail_delivery_job.rb#L30-L31) suggested it was specifically for use with exception handling but perhaps that comment can be removed.

Reply:
It is used to any code that can be executed before the arguments are deserialzied, or that could fail.

Reply:
Thanks. I'll update to use `mailer_class` then, and adjust the comment for clarity.